src/base/bittorrent/sessionstatus.h

```diff
@@ -29,50 +29,43 @@
 #ifndef BITTORRENT_SESSIONSTATUS_H
 #define BITTORRENT_SESSIONSTATUS_H
 
-#include <libtorrent/session_status.hpp>
 #include <QtGlobal>
 
 namespace BitTorrent
 {
-    class SessionStatus
+    struct SessionStatus
     {
-    public:
-        SessionStatus(const libtorrent::session_status &nativeStatus);
```

Comment:
>My main concern is that long `#ifdef` is hardly better then a long `if`.

But you can't get rid of `#ifdef`'s, unlike `if`'s (you can only move it to another place, and it is not an option).